src/Middlewares/OptimizeImages.php

```diff
@@ -12,8 +12,14 @@ public function handle($request, Closure $next)
     {
         $optimizerChain = app(OptimizerChain::class);
 
-        collect($request->allFiles())->each(function (UploadedFile $file) use ($optimizerChain) {
-            $optimizerChain->optimize($file->getPathname());
+        collect($request->allFiles())->each(function ($file) use ($optimizerChain) {
+            if (is_array($file)) {
+                collect($file)->each(function (UploadedFile $one) use ($optimizerChain) {
+                    $optimizerChain->optimize($one->getPathname());
+                });
+            } else {
```

Comment:
Could you rewrite this so an `else` isn't needed?

Reply:
> it will give error cuz it will run after the loop has finished.

Reply:
If file isn't an array, make it an array so you'll only need one path for processing it.

Reply:
updated, however i found a strange behavior with `$request->allFiles()`,

- if you have a form with single file upload ex.`avatar`
- using `$request->allFiles()` will return empty array